lib/src/util/dart_type_utilities.dart

```diff
@@ -61,8 +61,8 @@ class DartTypeUtilities {
 
   /// Builds the list resulting from traversing the node in DFS and does not
   /// include the node itself.
-  static List<AstNode> traverseNodesInDFS(AstNode node) {
-    List<AstNode> nodes = [];
+  static LinkedHashSet<AstNode> traverseNodesInDFS(AstNode node) {
+    LinkedHashSet<AstNode> nodes = new LinkedHashSet();
```

Comment:
It was enough for it to be iterable, performance is the reason, I measured something back then and came to the conclusion that the gain was worth the change, the new one is backwards compatible.